vignettes/irecist.Rmd

```diff
@@ -0,0 +1,858 @@
+
+```{r}
+confirmation_period <- 21
+
+icbor_icr <- event_joined(
+  description = paste(
+    "Define complete response (iCR) for confirmed best overall response (iCBOR) as",
+    "iCR followed by iCR at least",
+    confirmation_period,
+    "days later and at most one NE in between"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "after",
+  first_cond = AVALC.join == "iCR" &
+    ADT.join >= ADT + confirmation_period,
+  condition = AVALC == "iCR" &
+    all(AVALC.join %in% c("iCR", "NE")) &
+    count_vals(var = AVALC.join, val = "NE") <= 1,
+  set_values_to = exprs(AVALC = "iCR")
+)
+
+icbor_ipr <- event_joined(
+  description = paste(
+    "Define partial response (iPR) for confirmed best overall response (iCBOR) as",
+    "iPR followed by iCR or iPR at least",
+    confirmation_period,
+    "28 days later, at most one NE in between, and no iPR after iCR"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "after",
+  first_cond = AVALC.join %in% c("iCR", "iPR") &
+    ADT.join >= ADT + confirmation_period,
+  condition = AVALC == "iPR" &
+    all(AVALC.join %in% c("iCR", "iPR", "NE")) &
+    count_vals(var = AVALC.join, val = "NE") <= 1 &
+    (
+      min_cond(
+        var = ADT.join,
+        cond = AVALC.join == "iCR"
+      ) > max_cond(var = ADT.join, cond = AVALC.join == "iPR") |
+        count_vals(var = AVALC.join, val = "iCR") == 0 |
+        count_vals(var = AVALC.join, val = "iPR") == 0
+    ),
+  set_values_to = exprs(AVALC = "iPR")
+)
+
+no_data_n <- event(
+  description = "Define no response for all patients in adsl (should be used as last event)",
+  dataset_name = "adsl",
+  condition = TRUE,
+  set_values_to = exprs(AVALC = "N"),
+  keep_source_vars = adsl_vars
+)
+
+no_data_missing <- event(
+  description = paste(
+    "Define missing response (MISSING) for all patients in adsl (should be used",
+    "as last event)"
+  ),
+  dataset_name = "adsl",
+  condition = TRUE,
+  set_values_to = exprs(AVALC = "MISSING"),
+  keep_source_vars = adsl_vars
+)
+```
+
+## Derive Confirmed Progressive Disease Parameter {#pd}
+
+Now that we have the input records prepared above with any
+company-specific requirements, we can start to derive new parameter
+records. For the parameter derivations, all values except those
+overwritten by `set_values_to` argument are kept from the earliest
+occurring input record fulfilling the required criteria.
+
+The function `admiral::derive_extreme_records()` can be used to find
+the date of first `iCPD`.
+
+```{r}
+adrs <- adrs %>%
+  derive_extreme_records(
+    dataset_ref = adsl,
+    dataset_add = adrs,
+    by_vars = exprs(STUDYID, USUBJID),
+    filter_add = PARAMCD == "OVR" & AVALC == "iCPD" & ANL01FL == "Y",
+    order = exprs(ADT, RSSEQ),
+    mode = "first",
+    exist_flag = AVALC,
+    set_values_to = exprs(
+      PARAMCD = "ICPD",
```

Comment:
In general, I would remove "I" prefix from PARAMCD and "Immune" from PARAM since PARCAT3 gives information about response criteria.

Reply:
keep as is